plugin/src/main/groovy/com/here/gradle/plugins/jobdsl/JobDslPlugin.groovy

```diff
@@ -2,35 +2,79 @@ package com.here.gradle.plugins.jobdsl
 
 import com.here.gradle.plugins.jobdsl.tasks.GenerateXmlTask
 import com.here.gradle.plugins.jobdsl.tasks.UpdateJenkinsTask
+
 import org.gradle.api.Plugin
 import org.gradle.api.Project
+import org.gradle.api.tasks.Copy
 
 /**
  * Gradle plugin configuration for the Job DSL plugin. Configures tasks, extensions, and source sets.
  */
 class JobDslPlugin implements Plugin<Project> {
 
     @Override
+    @SuppressWarnings('DuplicateMapLiteral')
     void apply(Project project) {
         project.extensions.create('jobdsl', JobDslPluginExtension, project)
 
-        def execTasks = []
-        execTasks += project.task('dslGenerateXml', type: GenerateXmlTask)
-        execTasks += project.task('dslUpdateJenkins', type: UpdateJenkinsTask)
-
         project.plugins.apply('groovy')
 
-        execTasks.each { task ->
-            task.dependsOn 'classes'
-        }
-
         project.sourceSets {
             jobdsl {
                 groovy {
                     srcDirs 'src/jobdsl'
                 }
             }
         }
+
+        project.configurations {
+            jenkinsPlugins {
+                description 'Jenkins plugins that are required to process the Job DSL scripts.'
+            }
+        }
+
+        project.repositories {
+            mavenCentral()
+
+            maven {
+                url 'https://repo.jenkins-ci.org/releases/'
+            }
+        }
+
+        project.dependencies {
+            jenkinsPlugins('org.jenkins-ci.plugins:job-dsl:1.66') {
+                exclude(module: 'groovy-all')
+            }
+            jenkinsPlugins('org.jenkins-ci.plugins:job-dsl:1.66@jar') {
```

Comment:
Doesn't this duplicate the above (as the default is `@jar`)?

Reply:
I just treid and only the @hpi dependency is required so I have removed both @jar dependencies.

Reply:
Turns out I was wrong, the @hpi dependency is required for the plugin to be loaded in the Jenkins instance, while the @jar dependencies are required for on the classpath of the Gradle exec task, so I have recovered them.

Reply:
Can we please get code comments for this, too?

Reply:
Done.